.github/workflows/changelog.yml

```diff
@@ -0,0 +1,24 @@
+name: Changelog
+
+on:
+  pull_request:
+    # should also be re-run when changing labels
+    types: [opened, reopened, labeled, unlabeled, synchronize]
+
+env:
+  FRAGMENT_NAME: "docs/changes/${{ github.event.number }}.*.rst"
+
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check for news fragment
+        if: ${{ ! contains( github.event.pull_request.labels.*.name, 'no-changelog-needed')}}
```

Comment:
This seems to not work. Check still fails in this PR

Reply:
? All ten checks passed, including their changelog check

Reply:
Ok, I see. That’s weird I re-ran it two times after you added the label and they both failed. Maybe it was just a bug in the GitHub app…

Reply:
![image](https://user-images.githubusercontent.com/44728015/209121402-22f4ca43-e5b8-43f5-8095-7c08cd0f33c6.png)

Reply:
Definitely a bug with the app

Reply:
1 failed, 10 passed? But there are only 10 checks...